### PR TITLE
Main search box

### PR DESCRIPTION
**Pull request by ShArPman13**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related trello card link

_Describe the source of requirement, like related issue link._

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### 📷 Screenshots

image

### CODE REVIEW

**Review by Zmicer-Yafremau — APPROVED** [review]

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,13 @@
 <app-header></app-header>
 <div class="wrapper">
-  <router-outlet></router-outlet>
+  <div class="content-wrapper">
+    <router-outlet></router-outlet>
+    <app-main-search-box></app-main-search-box>
```
Comment (kubarenatioN) [inline src/app/app.component.html]: Please contact @LilithPrimary and discuss refactoring for this block + plan the merge process of our features

File: src/app/components/main-search-box/main-search-box.component.html [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    <mat-radio-button color="primary" value="oneway" (change)="typeOfTrip($event)">
+      One Way
+    </mat-radio-button>
+  </mat-radio-group>
+
+  <div class="trip-wrapper">
+    <mat-form-field class="from-input" appearance="outline">
+      <mat-label>From</mat-label>
+      <mat-select class="select" formControlName="from">
+        <mat-option *ngFor="let airport of airports" class="option" [value]="airport[1]">
```
Comment (kubarenatioN) [inline src/app/components/main-search-box/main-search-box.component.html]: Should it be `[value]="airport[0]"`?

File: src/app/components/main-search-box/main-search-box.component.html [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    <mat-form-field class="passengers-input" appearance="outline">
+      <mat-label>Passengers</mat-label>
+      <mat-select class="select">
+        <mat-select-trigger>
+          <span *ngIf="passengers.adults > 0"> Adults {{ passengers.adults }} </span>
+          <span *ngIf="passengers.children > 0"> Children {{ passengers.children }} </span>
+          <span *ngIf="passengers.infants > 0"> Infants {{ passengers.infants }} </span>
+        </mat-select-trigger>
+
+        <mat-option value="adult">
```
Comment (kubarenatioN) [inline src/app/components/main-search-box/main-search-box.component.html]: This option template repeats 3 times - for adult, child and infant. 
I suggest you create separate component and wrap this `<mat-option>`. 
Then, you will just need pass simple config with labels ("Adult", "14+ years", ...) and use `@Output` emitters to notify parent form about changes in counters. 

So the final template will look like this:
```
<app-passengers-option 
 [config]="adultsPassengersConfig" 
 (onChange)="handlePassengersChange('adult', $event)">
```

**Review by kubarenatioN — CHANGES_REQUESTED** [review]
Good job! 
Please, check comments and do refactoring.

File: src/app/components/main-search-box/main-search-box.component.html (hunk 3 above)
Comment (ShArPman13) [reply src/app/components/main-search-box/main-search-box.component.html]: done, please check

File: src/app/app.component.html (hunk 1 above)
Comment (ShArPman13) [reply src/app/app.component.html]: done

File: src/app/components/main-search-box/main-search-box.component.html [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          <span *ngIf="passengers.infants > 0"> Infants {{ passengers.infants }} </span>
+        </mat-select-trigger>
+
+        <mat-option [value]="passengersConfig.adults">
+          <app-passengers-option
+            [config]="passengersConfig.adults"
+            [passengers]="passengers"
+            (passengersEvent)="handlePassengersChange($event)"
+          />
+        </mat-option>
```
Comment (kubarenatioN) [inline src/app/components/main-search-box/main-search-box.component.html]: next time add `*ngFor` for config

File: src/app/app-routing.module.ts [hunk 5]
```diff
@@ -11,6 +12,10 @@ const routes: Routes = [
     loadChildren: () =>
       import('./user-profile/user-profile.module').then((m) => m.UserProfileModule),
   },
+  {
+    path: '',
+    component: MainSearchBoxComponent,
```
Comment (kubarenatioN) [inline src/app/app-routing.module.ts]: For empty route we need to create wrapper component and place in it all other components for main app page. Please replace component for this route by `MainComponent` and place in `MainComponent` your feature component, like

_main.component.html_
```
<app-main-search-box>
```

**Review by kubarenatioN — CHANGES_REQUESTED** [review]
Please take a look on comments. In general, everything done nice!

File: src/app/app-routing.module.ts (hunk 5 above)
Comment (ShArPman13) [reply src/app/app-routing.module.ts]: done

**Review by kubarenatioN — APPROVED** [review]
Well done!

**Pull request merged**